### PR TITLE
[Serve][Doc] Split the scaling and resource allocation into two pages

### PR DESCRIPTION
**Pull request by sihanwang41**
Signed-off-by: Sihan Wang <sihanwang41@gmail.com>

## Why are these changes needed?

Easy to find the gpu & cpu support in ray serve

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/serve/resource-allocation.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+(serve-cpus-gpus)=
+
+# CPU & GPU support and Customized resource
```
Comment (architkulkarni) [inline doc/source/serve/resource-allocation.md:3]:
```suggestion
# CPU and GPU support and custom resources
```

File: doc/source/serve/resource-allocation.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+This guide helps you to:
+- allocate hardware resources for each deployment
+- allocate fractional CPU & GPU resources
+- allocate customized resources
```
Comment (architkulkarni) [inline doc/source/serve/resource-allocation.md:8]:
```suggestion
- allocate custom resources
```

File: doc/source/serve/resource-allocation.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+## Allocate resources
```
Comment (architkulkarni) [inline doc/source/serve/resource-allocation.md:10]:
```suggestion
## Allocating resources
```

File: doc/source/serve/user-guide.md [hunk 4]
```diff
@@ -5,7 +5,8 @@ If you’re new to Ray Serve, we recommend starting with the [Ray Serve Quick St
 
 This user guide will help you navigate the Ray Serve project and show you how to achieve several key tasks:
 - [HTTP Handling](http-guide)
-- [Scaling and Resource Allocation](scaling-and-resource-allocation)
+- [Scaling](scaling)
+- [CPU & GPU support](serve-cpus-gpus)
```
Comment (architkulkarni) [inline doc/source/serve/user-guide.md:9]:
```suggestion
- [CPU and GPU support](serve-cpus-gpus)
```

File: doc/source/serve/resource-allocation.md (hunk 1 above)
Comment (architkulkarni) [reply doc/source/serve/resource-allocation.md:3]: "and" is preferred over "&" here (I had to google it).  Also changing to "custom resources" for consistency with the rest of the doc

**Review by architkulkarni — APPROVED** [review]
Thanks, just had a few nits in the text.  @richardliaw can you give ray-docs approval?

**Review by richardliaw — APPROVED** [review]

**Review by simon-mo — APPROVED** [review]

**Comment by simon-mo** [comment]
>  /ray/doc/source/serve/key-concepts.md:124: WARNING: 'myst' reference target not found: scaling-and-resource-allocation


doc build failed

**Pull request closed without merging**